### ISSUE TITLE
Pinentry

### ISSUE THREAD

**Issue by toastal** [soywod/himalaya#377]
Tools like `mbsync` and `vdirsyncer`, when `pass` is called, if it can't the TUI `pinentry` will show up because the keychain isn't opened. This can unknowingly cause confusing errors like 535 (GitHub issue #294) or make the application in general fail to load.

**Comment by soywod** [comment]
Not sure to understand. What do you propose exactly?

**Comment by kLabz** [comment]
@soywod when called without a display environment (for example via ssh), `pass` can use a TUI for `pinentry` to ask for passphrase. Himalaya can't, so when we need to enter the passphrase the command just fails.

There is a workaround: while `himalaya list` will fail over ssh when passphrase is needed, `GPG_TTY=$(tty) himalaya list` will work (= prompt for passphrase from a TUI). Haven't found a way to make it work nicely from within `(imap|smtp)-passwd-cmd` itself yet.

Maybe himalaya itself could/should set `GPG_TTY` env?

**Comment by toastal** [comment]
Phrased much better than I could do. Thanks @kLabz :sweat_smile:

**Comment by soywod** [comment]
I see. I don't think setting the `GPG_TTY` is the good approach, himalaya should not be aware of gpg. I checked the source code of `isync`, `vdirsyncer` and `aerc`, I did not find anything related to the `GPG_TTY` env var. I must miss sth. This point needs to be dug carefully. I don't want to ship a quick'n dirty patch by setting the var, especially that the [gpg doc](https://www.gnupg.org/documentation/manuals/gnupg/Invoking-GPG_002dAGENT.html) explicitly mention to add the command in your `.bashrc` or equivalent:

> You should always add the following lines to your .bashrc or whatever initialization file is used for all shell invocations:
> ```bash
> GPG_TTY=$(tty)
> export GPG_TTY
> ```

But I'm not against a smart solution to avoid confusing error messages!
